spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/policy/PolicyEngine.java

```diff
@@ -22,38 +22,56 @@
 
 /**
  * Evaluates policies.
+ *
+ * A policy scope is a visibility and semantic boundary for a {@link Rule}. A rule binding associates a rule type (e.g. 'geospatial') with a scope identified by a key, thereby
+ * making a policy visible in a scope. Rule and constraint functions can be bound to one or more scopes, limiting the semantics they implement to the scope they are
+ * registered with.
+ *
+ * Scopes are hierarchical and delimited by {@link #DELIMITER}. Functions bound to parent scopes will be inherited in child scopes.
  */
 public interface PolicyEngine {
 
     /**
-     * Evaluates the given policy for an agent.
+     * Wildcard denoting all scopes.
+     */
+    String ALL_SCOPES = "*";
+
+    /**
+     * Scope delimiter.
+     */
+    String DELIMITER = ".";
```

Comment:
I would prefer it as a pattern that encapsulates the creation of the full delimited scope:
~~~
String SCOPE_PATTERN = "%s.";
~~~

And then everywhere a scope should be built:
~~~
String fullScope = String.format(SCOPE_PATTERN , myScope);
~~~

So that if one day we want to change the logic to construct the full scope, we only change it in one place.

Reply:
How would we handle a scope specified as "foo.bar.baz"?

Reply:
Hum did not think about it... Nevermind it's details. Ok for me.